streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -213,6 +216,18 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
     zipWith prioritizes failure $zipWithPrioritizesFailure
   """
 
+  def processRunCollect = {
+    def loop[E, A](effect: IO[_, A], ref: Ref[List[A]]): IO[Nothing, List[A]] =
+      (effect.flatMap(a => ref.update(a :: _)) *> loop(effect, ref)).catchAll(_ => ref.get)
```

Comment:
you can use ZIO#forever to make this a bit simpler

Reply:
Done.